python-sdk/nuscenes/eval/tracking/tests/test_algo.py

```diff
@@ -0,0 +1,190 @@
+        timestamp_boxes_pred[2][0].tracking_id = 'tb'
+        tracks_pred = {'scene-1': timestamp_boxes_pred}
+
+        # Accumulate metrics.
+        ev = TrackingEvaluation(tracks_gt, tracks_pred, class_name, cfg.dist_fcn_callable,
+                                cfg.dist_th_tp, cfg.min_recall, num_thresholds=TrackingMetricData.nelem,
+                                verbose=verbose)
+        md = ev.accumulate()
+
+        # Check outputs.
+        assert md.tp[5] == 2
+        assert md.fp[5] == 0
+        assert md.fn[5] == 0
+        assert md.lgd[5] == 0  # TODO: Fix this. IDS should count as misses in LGD and TID.
+        assert md.tid[5] == 0
+        assert md.frag[5] == 0
+        assert md.ids[5] == 2  # One wrong id leads to 2 identity switches.
+
+    def test_drop_gt(self):
+
+        # Get config.
+        cfg = config_factory('tracking_nips_2019')
+
+        # Define inputs.
+        class_name, tracks_gt = TestAlgo.single_scene()
+        verbose = False
+
+        # Remove one GT box.
+        timestamp_boxes_pred = copy.deepcopy(tracks_gt['scene-1'])
+        tracks_gt['scene-1'][1] = []
+        tracks_pred = {'scene-1': timestamp_boxes_pred}
+
+        # Accumulate metrics.
+        ev = TrackingEvaluation(tracks_gt, tracks_pred, class_name, cfg.dist_fcn_callable,
+                                cfg.dist_th_tp, cfg.min_recall, num_thresholds=TrackingMetricData.nelem,
+                                verbose=verbose)
+        md = ev.accumulate()
+
+        # Check outputs.
+        assert np.all(md.tp == 3)
+        assert np.all(md.fp == 1)
+        assert np.all(md.fn == 0)
+        assert np.all(md.lgd == 0)
+        assert np.all(md.tid == 0)
+        assert np.all(md.frag == 0)
+        assert np.all(md.ids == 0)
+
+    def test_drop_gt_interpolate(self):
+
+        # Get config.
+        cfg = config_factory('tracking_nips_2019')
+
+        # Define inputs.
+        class_name, tracks_gt = TestAlgo.single_scene()
+        verbose = False
+
+        # Remove one GT box.
+        timestamp_boxes_pred = copy.deepcopy(tracks_gt['scene-1'])
+        tracks_gt['scene-1'][1] = []
+        tracks_pred = {'scene-1': timestamp_boxes_pred}
+
+        # Interpolate to "restore" dropped GT.
+        tracks_gt['scene-1'] = interpolate_tracks(tracks_gt['scene-1'])
+
+        # Accumulate metrics.
+        ev = TrackingEvaluation(tracks_gt, tracks_pred, class_name, cfg.dist_fcn_callable,
+                                cfg.dist_th_tp, cfg.min_recall, num_thresholds=TrackingMetricData.nelem,
+                                verbose=verbose)
+        md = ev.accumulate()
+
+        # Check outputs.
+        assert np.all(md.tp == 4)
+        assert np.all(md.fp == 0)
+        assert np.all(md.fn == 0)
+        assert np.all(md.frag == 0)
+        assert np.all(md.lgd == 0)
+        assert np.all(md.tid == 0)
+        assert np.all(md.ids == 0)
+
+
+if __name__ == '__main__':
```

Comment:
Any particular reason you list the tests explicitly rather than doing `unittest.main()`.

Reply:
No. This sounds like a better alternative.